src/addr_validate.rs

```diff
@@ -42,56 +54,50 @@ fn create_pipe() -> nix::Result<(i32, i32)> {
 }
 
 fn open_pipe() -> nix::Result<()> {
-    MEM_VALIDATE_PIPE.with(|pipes| {
-        let mut pipes = pipes.borrow_mut();
+    let pipes = unsafe { &mut *MEM_VALIDATE_PIPE.get() };
```

Comment:
We should add synchronization in multi-thread cases. Otherwise, there may be many problems.

For example, on a weak-memory architecture, when the value can be read is indefinite. It's also possible that `pipes[0]` and `pipes[1]` don't pair. Or, creating fd leaks because some fds are silently overwritten by other threads.

Reply:
Ah, concurrent access may be impossible in a signal handler...(?)

Reply:
Signal handlers don't be called concurrently, so I think an atomic fence should be enough.

Reply:
Thanks :+1: . Fixed.